include/eve/module/core/regular/impl/scatter.hpp

```diff
@@ -18,25 +18,40 @@ namespace eve::detail
   template<typename T, typename Idx, typename Ptr, callable_options O>
   EVE_FORCEINLINE void scatter_(EVE_REQUIRES(cpu_), O const& o, T const& v, Ptr p, Idx const& idx)
   {
-    // Retrieve element to scatter
-    auto se   = store_equivalent(o[condition_key],v,p);
-
-    // Extract the pointer from a potential aligned_ptr
-    auto base = unalign(get<2>(se));
+    // Aggregate can be optimised if p is sort of random access and there is no mask
+    // Mask would require slice on ignore_*** which we don't do atm
+    if constexpr(has_aggregated_abi_v<T> && requires{ p + 1; } && !O::contains(condition_key))
```

Comment:
all pointers are random access. we don't do non random access.

Reply:
That's true